src/proofs/weighted_inner_product.rs

```diff
@@ -0,0 +1,915 @@
+#![allow(non_snake_case)]
+
+/*
+
+Copyright 2018 by Kzen Networks
+
+This file is part of bulletproof library
+(https://github.com/KZen-networks/bulletproof)
+
+bulletproof is free software: you can redistribute
+it and/or modify it under the terms of the GNU General Public
+License as published by the Free Software Foundation, either
+version 3 of the License, or (at your option) any later version.
+
+@license GPL-3.0+ <https://github.com/KZen-networks/bulletproof/blob/master/LICENSE>
+*/
+
+// based on the paper: https://eprint.iacr.org/2017/1066.pdf
```

Comment:
https://eprint.iacr.org/2020/735.pdf. 
I would also add a one liner on the difference

Reply:
Done.